tests/system/src/test/java/test/com/sun/prism/impl/NGNodeDirtyFlagTest.java

```diff
@@ -0,0 +1,188 @@
+            primaryStage.show();
+
+            launchLatch.countDown();
+        }
+    }
+
+    @BeforeClass
+    public static void setupOnce() {
+        Util.launch(launchLatch, MyApp.class);
+        assertEquals(0, launchLatch.getCount());
+    }
+
+    @AfterClass
+    public static void teardownOnce() {
+        Util.shutdown();
+    }
+
+    @Test
+    public void testNGNodesNotDirty() throws InterruptedException {
+        ObjectProperty<Color> lineColor = new SimpleObjectProperty<>(Color.DARKGREEN);
+        ObjectProperty<Color> circleColor = new SimpleObjectProperty<>(Color.DARKGREEN);
+
+        StackPane root = myApp.root;
+
+        runAndWait(() -> {
+            var contents = new HBox();
+            contents.setSpacing(10);
+            contents.setPadding(new Insets(10));
+            contents.getChildren().add(contentElement("L", lineColor, circleColor));
+            contents.getChildren().add(contentElement("R", lineColor, circleColor));
+            root.getChildren().add(contents);
+
+            Pane sideArea = createSideArea();
+            StackPane.setAlignment(sideArea, Pos.CENTER_RIGHT);
+            root.getChildren().add(sideArea);
+        });
+
+        Thread.sleep(500);
+
+        for (int i = 0; i < 5; i++) {
+            Platform.runLater(() -> lineColor.set(Color.LIGHTGREEN));
+            Thread.sleep(300);
+            Platform.runLater(() -> circleColor.set(Color.LIGHTGREEN));
+            Thread.sleep(300);
+
+            checkLineColor(root, lineColor.get());
+
+            Platform.runLater(() -> lineColor.set(Color.DARKGREEN));
+            Thread.sleep(300);
+            Platform.runLater(() -> circleColor.set(Color.DARKGREEN));
+            Thread.sleep(300);
+
+            checkLineColor(root, lineColor.get());
+        }
+
+    }
+
+    private void checkLineColor(StackPane root, Color expected) {
+        runAndWait(() -> {
+            checkColor(root.lookup("#Line-L"), expected);
+            checkColor(root.lookup("#Line-R"), expected);
+        });
+    }
+
+    private void checkColor(Node node, Color expected) {
+        Robot robot = new Robot();
+        Bounds screenBounds = node.localToScreen(node.getBoundsInLocal());
+        WritableImage image = robot.getScreenCapture(null, screenBounds.getMinX(), screenBounds.getMinY(), 100, 100);
+        Assert.assertEquals("A node was not rendered properly. Wrong color found", expected, image.getPixelReader().getColor(1, 1));
+    }
+
+    private void runAndWait(Runnable action) {
+        try {
+            CompletableFuture.runAsync(action, Platform::runLater).get(5000, TimeUnit.MILLISECONDS);
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof AssertionError error) {
+                throw error;
+            } else {
+                throw new RuntimeException(e);
+            }
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
```

Comment:
Please use the `Util.runAndWait()` instead of adding this method. So this method can be removed and all calls can be changed to Util.runAndWait().
Please do remove any imports that become unused afterwards.

Reply:
Done